modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -123,8 +123,10 @@ class HdkOnNativeDataframe(PandasDataframe):
     _table_cols : list of str
         A list of all frame's columns. It includes index columns if any. Index
         columns are always in the head of the list.
-    _index_cache : pandas.Index or None
+    _index_cache : pandas.Index, callable or None
         Materialized index of the frame or None when index is not materialized.
+        If ``callable() -> (pandas.Index, list or None)`` type then the
+        calculation will be done in `__init__`.
```

Comment:
```suggestion
        If ``callable() -> (pandas.Index, list of row lengths or None)`` type,
        then the calculation will be done in `__init__`.
```

Reply:
The list type is the row lengths that are stored in it. I believe that only types should be specified in the signature of the callable.

Reply:
It is not clear what is the list. "list of row lengths" or just "row lengths" looks much clearer to me.